modules/core/src/test/java/org/apache/ignite/internal/processors/cache/binary/BinaryMetadataMoveLegacyFolderTest.java

```diff
@@ -0,0 +1,225 @@
+import org.apache.ignite.configuration.DataRegionConfiguration;
+import org.apache.ignite.configuration.DataStorageConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.binary.BinaryMetadata;
+import org.apache.ignite.internal.binary.BinaryTypeImpl;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * Test for moving binary metadata folder to PDS
+ */
+public class BinaryMetadataMoveLegacyFolderTest extends GridCommonAbstractTest {
+    /** {@inheritDoc} */
+    @Before
+    @Override public void beforeTest() throws Exception {
+        super.beforeTest();
+
+        stopAllGrids();
+
+        cleanPersistenceDir();
+    }
+
+    /** {@inheritDoc} */
+    @After
+    @Override public void afterTest() throws Exception {
+        super.afterTest();
+
+        stopAllGrids();
+
+        cleanPersistenceDir();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration configuration = super.getConfiguration(igniteInstanceName);
+
+        configuration.setConsistentId(UUID.randomUUID());
+
+        DataStorageConfiguration dsCfg = new DataStorageConfiguration();
+
+        configuration.setDataStorageConfiguration(dsCfg);
+
+        dsCfg.setDefaultDataRegionConfiguration(
+            new DataRegionConfiguration()
+                .setPersistenceEnabled(true));
+
+        return configuration;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected IgniteEx startGrid(IgniteConfiguration cfg) throws Exception {
+        IgniteEx grid = super.startGrid(cfg);
+        grid.active(true);
+        return grid;
+    }
+
+    /**
+     * Get byte representation of simple binary type with one field
+     *
+     * @param typeName  Type name.
+     * @param fieldName Field name.
+     */
+    private byte[] createBinaryType(String typeName, String fieldName) throws Exception {
+        IgniteConfiguration configuration = getConfiguration();
+
+        IgniteEx grid = startGrid(configuration);
+
+        IgniteBinary binary = grid.binary();
+
+        BinaryObjectBuilder builder = binary.builder("TestBinaryType");
+
+        builder.setField(fieldName, 1);
+
+        builder.build();
+
+        BinaryMetadata metadata = ((BinaryTypeImpl)binary.type(typeName)).metadata();
+
+        byte[] marshalled = U.marshal(grid.context(), metadata);
+
+        stopAllGrids();
+
+        cleanPersistenceDir();
+
+        return marshalled;
+    }
+
+    /**
+     * Test that binary_meta directory, that was previously located in workdir's root, is successfully moving to PDS
+     * folder.
+     */
+    @Test
+    public void testBinaryMetadataDirectoryMigration() throws Exception {
```

Comment:
You should use the `ignite-compatibility` framework to test that a new Ignite version able to start on the old persistence directory. 
See: https://issues.apache.org/jira/browse/IGNITE-5732

Reply:
Added compatibility test in newest commit (testing against version 2.8.0)